SteamRPC.js

```diff
@@ -110,25 +107,64 @@ async function pollSteamPresence(steamUserId, profiles) {
-                        discordRPCClient.user?.setActivity(translatedDiscordRPC);
-                    }
-                    catch (codeErr) {
-                        throw new Error(`A code error has occured in the game profile for '${profile.title}'!`, {cause: codeErr});
+            gameStatus = resJson.in_game.name;
+
+            if(resJson.in_game?.rich_presence === "")
+                debugLine = "There is no RPC status for this game.";
+            else {
+                if (resJson.in_game?.logo && resJson.in_game?.rich_presence) {
+                    let currentAppId = resJson.in_game.logo.split("/apps/")[1].split("/")[0];
+                    let currentSteamStatus = resJson.in_game.rich_presence;
+                    debugLine = "Current Steam RPC Status: " + resJson.in_game.rich_presence;
+
+                    if (profiles.hasOwnProperty(currentAppId)) {
+                        let profile = profiles[currentAppId];
+                        try {
+                            gameStatus = profile.title;
+                            let clientId;
+
+                            if(profile.clientId !== undefined)
+                                clientId = profile.clientId;
+                            else
+                                clientId = config.default_client_id;
+    
+                            if(discordRPCClient === null) 
+                                initRPC(clientId);
+                            else if(discordRPCClient.clientId !== clientId) {
+                                discordRPCClient.login({clientId});
+                            }
+    
+                            let translatedDiscordRPC = profile.translateSteamPresence(currentSteamStatus, discordRPCClient);
+
+                            if (typeof translatedDiscordRPC !== "object") 
+                                throw `Profile returned '${typeof translatedDiscordRPC}' instead of an object.`;
+    
+                            if(translatedDiscordRPC.startTimestamp) {
+                                if(lastTimestamp !== null) translatedDiscordRPC.startTimestamp = lastTimestamp;
+                                else lastTimestamp = translatedDiscordRPC.startTimestamp;
+                            }
+
+                            else {
+                                lastTimestamp = null;
+                            }
+
+                            translatedDiscordRPC.largeImageText = `SteamRPC - v${version}`;
+                            discordRPCClient.user?.setActivity(translatedDiscordRPC);
+                        }
+                        catch (codeErr) {
+                            throw new Error(`A code error has occured in the game profile for '${profile.title}'!`, {cause: codeErr});
+                        }
                     }
                 }
             }
         }
+
+        else {
+            if(discordRPCClient !== null) {
+                discordRPCClient.destroy();
```

Comment:
Destroying is asynchronous. Not that it matters much in this case since there's a pretty decent interval after this, but I thought I'd note it.
```suggestion
                await discordRPCClient.destroy();
```